### PR TITLE
Upgrade vLLM from v0.5 series to v0.6

### PR DESCRIPTION
**Pull request by butsugiri**
This PR upgrades vLLM version from v0.5 series to v0.6.

### CODE REVIEW

**Comment by Ktakuya332C** [comment]
Do you mind if I ask you to run unit-tests on a machine with GPUs ? Since all the test cases for vllm are disabled on a machine without GPUs (which implies all tests testing vllm are skipped in CI runs), I would like to know whether they will pass or not.
https://github.com/sbintuitions/flexeval/blob/64ae201aa7551af88643c3febf872542a15b9174/tests/core/language_model/test_vllm_model.py#L12

**Comment by butsugiri** [comment]
Ah, good catch! Thanks for pointing that out. I'll check it.

File: pyproject.toml [hunk 1]
```diff
@@ -29,7 +29,7 @@ sacrebleu = {extras = ["ja"], version = "^2.4.1"}
 jiwer = "^3.0.4"
 openai = "^1.52.2"
 google-api-python-client = "^2.131.0"
-vllm = {version = "^0.5.5", optional = true }
+vllm = {version = "^0.6.4", optional = true }
```
Comment (jsato8094) [inline pyproject.toml:32]:
```suggestion
vllm = {version = "^0.6.4.post1", optional = true }
```
I think it is better to specify `0.6.4.post1` which is bug fix version
https://github.com/vllm-project/vllm/releases

**Comment by butsugiri** [comment]
```
================================================================================== short test summary info ==================================================================================
FAILED tests/core/language_model/test_hf_lm.py::test_batch_compute_log_probs_is_not_affected_by_batch - assert -48.1735 == -48.1733
FAILED tests/core/language_model/test_vllm_model.py::test_batch_generate_chat_response - ValueError: Cannot use chat template functions because tokenizer.chat_template is not set and no template argument was passed! For information about writing templates and setting the t...
FAILED tests/core/language_model/test_vllm_model.py::test_if_gen_kwargs_work_as_expected - torch.distributed.DistNetworkError: The client socket has timed out after 600000ms while trying to connect to (127.0.0.1, 44327).
ERROR tests/core/language_model/test_vllm_model.py::test_if_custom_chat_template_is_given - torch.distributed.DistNetworkError: The client socket has timed out after 600000ms while trying to connect to (127.0.0.1, 37141).
ERROR tests/core/language_model/test_vllm_model.py::test_if_stop_sequences_work_as_expected - torch.distributed.DistNetworkError: The client socket has timed out after 600000ms while trying to connect to (127.0.0.1, 46071).
ERROR tests/core/language_model/test_vllm_model.py::test_batch_compute_chat_log_probs - torch.distributed.DistNetworkError: The client socket has timed out after 600000ms while trying to connect to (127.0.0.1, 46071).
ERROR tests/core/language_model/test_vllm_model.py::test_compute_chat_log_probs - torch.distributed.DistNetworkError: The client socket has timed out after 600000ms while trying to connect to (127.0.0.1, 46071).
======================================================= 3 failed, 396 passed, 3 skipped, 16 warnings, 4 errors in 3915.96s (1:05:15) ========================================================
```

It turns out that several tests are failing. I will take a look if I can fix them.

**Comment by butsugiri** [comment]
Initializing multiple vLLM instances at once was causing a hang.

To fix the hang, the following modifications are needed.

- Dividing current vLLM tests into multiple files
- Destroying vLLM instance as soon as the test is done
- Instantiating vLLM with `disable_custom_all_reduce=True` 
  - as in https://github.com/vllm-project/vllm/issues/1908#issuecomment-2451005641
  - without this option, the hang persists even if we destroy the instance.

**Comment by butsugiri** [comment]
Now the vLLM test is passing. Asking for a review.

**Review by Ktakuya332C — APPROVED** [review]

**Pull request merged**